src/utils/viewer/viewer.cpp

```diff
@@ -330,6 +335,10 @@ Key Commands:
 
   const int defaultAgentId_ = 0;
   esp::agent::Agent::ptr defaultAgent_ = nullptr;
+  std::string lastAddedObjectName_ = "";
```

Comment:
All the new code in the viewer.cpp is for your demo only. Perhaps you do not have to commit it in with this PR.
So this PR should include probably everything expect the viewer.cpp.
Then you create a new branch on top of this branch, `vauduong:decouple-sensors` (careful, not on top of the `master`) and put the modified `viewer.cpp` in that new branch (no PR is needed.)